htdocs/product/stock/stockatdate.php

```diff
@@ -353,164 +359,187 @@
-}
-if (GETPOSTINT('datemonth') > 0) {
-	$param .= '&datemonth='.GETPOSTINT('datemonth');
-}
-if (GETPOSTINT('dateyear') > 0) {
-	$param .= '&dateyear='.GETPOSTINT('dateyear');
-}
 
-// TODO Move this into the title line ?
-print_barre_liste('', $page, $_SERVER["PHP_SELF"], $param, $sortfield, $sortorder, '', $num, $nbtotalofrecords, 'stock', 0, '', '', $limit, 0, 0, 1);
+	print '<div class="inline-block valignmiddle">';
+	print '<input type="submit" class="button" name="valid" value="'.$langs->trans('Refresh').'">';
+	print '</div>';
 
-print '<div class="div-table-responsive">'; // You can use div-table-responsive-no-min if you don't need reserved height for your table
-print '<table class="liste centpercent">';
+	//print '</form>';
 
-$stocklabel = $langs->trans('StockAtDate');
-if ($mode == 'future') {
-	$stocklabel = $langs->trans("VirtualStockAtDate");
-}
+	$param = '';
+	if (!empty($contextpage) && $contextpage != $_SERVER["PHP_SELF"]) {
+		$param .= '&contextpage='.urlencode($contextpage);
+	}
+	if ($limit > 0 && $limit != $conf->liste_limit) {
+		$param .= '&limit='.((int) $limit);
+	}
+	$param .= '&mode='.$mode;
+	$param_warehouse = '';
+	if (!empty($search_fk_warehouse)) {
+		foreach ($search_fk_warehouse as $val) {
+			$param_warehouse .= '&search_fk_warehouse[]='.$val;
+		}
+		$param .= $param_warehouse;
+	}
+	if ($productid > 0) {
+		$param .= '&productid='.(int) $productid;
+	}
+	if (GETPOST('dateday', 'int') > 0) {
+		$param .= '&dateday='.GETPOST('dateday', 'int');
+	}
+	if (GETPOST('datemonth', 'int') > 0) {
+		$param .= '&datemonth='.GETPOST('datemonth', 'int');
+	}
+	if (GETPOST('dateyear', 'int') > 0) {
+		$param .= '&dateyear='.GETPOST('dateyear', 'int');
+	}
 
-print '<input type="hidden" name="token" value="'.newToken().'">';
-print '<input type="hidden" name="sortfield" value="'.$sortfield.'">';
-print '<input type="hidden" name="sortorder" value="'.$sortorder.'">';
-print '<input type="hidden" name="type" value="'.$type.'">';
-print '<input type="hidden" name="mode" value="'.$mode.'">';
-
-// Fields title search
-print '<tr class="liste_titre_filter">';
-print '<td class="liste_titre"><input class="flat" type="text" name="search_ref" size="8" value="'.dol_escape_htmltag($search_ref).'"></td>';
-print '<td class="liste_titre"><input class="flat" type="text" name="search_nom" size="8" value="'.dol_escape_htmltag($search_nom).'"></td>';
-print '<td class="liste_titre"></td>';
-print '<td class="liste_titre"></td>';
-print '<td class="liste_titre"></td>';
-if ($mode == 'future') {
+	// TODO Move this into the title line ?
+	print_barre_liste('', $page, $_SERVER["PHP_SELF"], $param, $sortfield, $sortorder, '', $num, $nbtotalofrecords, 'stock', 0, '', '', $limit, 0, 0, 1);
+
+	print '<div class="div-table-responsive">'; // You can use div-table-responsive-no-min if you don't need reserved height for your table
+	if ($num) {
+		print "<p><a href=\"stockatdate.csv?sortfield=$sortfield&sortorder=$sortorder&type=$type&mode=$mode".
```

Comment:
A dynamic page must end with .php, using another extension will generate troubles when using proxy or cache server.
You can add the parameter  &output=file  to provide the information you want a download.